spinta/datasets/commands/check.py

```diff
@@ -1,14 +1,31 @@
+import re
+
 from spinta import commands
-from spinta.components import Context
+from spinta.components import Context, Config
 from spinta.datasets.components import Dataset, Resource
+from spinta.exceptions import InvalidName
 
 
 @commands.check.register()
 def check(context: Context, dataset: Dataset):
     for resource in dataset.resources.values():
         commands.check(context, resource)
 
+    config: Config = context.get('config')
+
+    if config.check_names:
+        name = dataset.name
+        is_lowercase = re.compile(r'^[a-z0-1_]*(/[a-z0-1_]*)+$')
```

Comment:
Move this to module level. Also, dataset name can't start with `_`, this is for reserved names. There are some reserved names, but these names should be listed and checked against know reserved names.

Also each part separated with `/` in dataset name should start with `[a-z]` and only then, second ant other symbols can be `[a-z0-9_]`. Dataset name can't start with `/`, so in your regexp first block should end with `+` and second with `*`.